src/coreComponents/constitutive/solid/CoupledSolid.hpp

```diff
@@ -0,0 +1,180 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+
+/**
+ * @file CoupledSolid.hpp
+ */
+
+#ifndef GEOSX_CONSTITUTIVE_SOLID_COUPLEDSOLID_HPP_
+#define GEOSX_CONSTITUTIVE_SOLID_COUPLEDSOLID_HPP_
+
+#include "constitutive/ConstitutiveBase.hpp"
+#include "constitutive/solid/SolidBase.hpp"
+
+namespace geosx
+{
+namespace constitutive
+{
+
+template< typename SOLID_TYPE,
+          typename PORO_TYPE,
+          typename PERM_TYPE >
+class CoupledSolidUpdates
+{
+public:
+  /**
+   * @brief Constructor
+   */
+  CoupledSolidUpdates( SOLID_TYPE const & solidModel,
+                       PORO_TYPE const & porosityModel,
+                       PERM_TYPE const & permModel ):
+    m_solidUpdate( solidModel.createKernelUpdates() ),
+    m_porosityUpdate( porosityModel.createKernelUpdates() ),
+    m_permUpdate( permModel.createKernelWrapper() )
+  {}
+
+  GEOSX_HOST_DEVICE
+  real64 getOldPorosity( localIndex const k,
+                         localIndex const q ) const
+  {
+    return m_porosityUpdate.getOldPorosity( k, q );
+  }
+
+  GEOSX_HOST_DEVICE
+  real64 getPorosity( localIndex const k,
+                      localIndex const q ) const
+  {
+    return m_porosityUpdate.getPorosity( k, q );
+  }
+
+protected:
+  typename SOLID_TYPE::KernelWrapper const m_solidUpdate;
+
+  typename PORO_TYPE::KernelWrapper const m_porosityUpdate;
+
+  typename PERM_TYPE::KernelWrapper const m_permUpdate;
+};
+
+
+
+/**
+ * @brief Class to represent a coupled solid model
+ */
+template< typename SOLID_TYPE,
+          typename PORO_TYPE,
+          typename PERM_TYPE >
+class CoupledSolid : public ConstitutiveBase
+{
+public:
+
+  /**
+   * @brief Constructor
+   * @param name Object name
+   * @param parent Object's parent group
+   */
+  CoupledSolid( string const & name, dataRepository::Group * const parent );
+
+  /// Destructor
+  virtual ~CoupledSolid() override;
+
+  /**
+   * @brief Catalog name
+   * @return Static catalog string
+   */
+  static string catalogName() { return string( "Poro" ) + SOLID_TYPE::m_catalogNameString; }
```

Comment:
"Poro" should be changed to something else?

Reply:
yes, it should. Although this is class is never instantiated so it does not really matter.